core/batch.go

```diff
@@ -48,6 +59,36 @@ func (db *DB) GetBatchByIndex(index uint64) (batch Batch, err error) {
 	return batch, nil
 }
 
-func (db *DB) CommitBatch(batch Batch) error {
-	return db.Instance.Update(batch).Error
+func (db *DB) CommitBatch(ID uint64) error {
+	batch, err := db.GetBatchByIndex(ID)
+	if err != nil {
+		return err
+	}
+
+	if batch.Status != BATCH_BROADCASTED && batch.Status == BATCH_COMMITTED {
```

Comment:
I think we can safely remove `batch.Status != BATCH_BROADCASTED`. 
- When `batch.Status != BATCH_COMMITTED` the whole condition evaluates false
- when `batch.Status == BATCH_COMMITTED`, `batch.Status != BATCH_BROADCASTED` evaluates true, so the whole condition evaluates true too.

```suggestion
	if batch.Status == BATCH_COMMITTED {
```

Reply:
Yeah I think your suggestion assumes that the status can be either BATCH_BROADCASTED or BATCH_COMMITTED while I was trying to protect from invalid status codes too, which I think is stupid🤐

Reply:
Fixing